tclx/eval/inference.py

```diff
@@ -0,0 +1,130 @@
+from transformers import AutoModelForCausalLM, AutoTokenizer
+import json
+import torch
+from torch.utils.data import Dataset, random_split
+from tqdm import tqdm
+from datasets import load_dataset
+from logger import Logger
+from accelerate import Accelerator
+import argparse
+from transformers import StoppingCriteria, StoppingCriteriaList
+import os
+from utils import make_rm, load_rm
+import random
+
+
+class StoppingCriteriaSub(StoppingCriteria):
+    def __init__(self, stops = []):
+      StoppingCriteria.__init__(self), 
+
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, stops = []):
+      self.stops = stops
+      for i in range(len(stops)):
+        self.stops = self.stops[i]
+
```

Comment:
`StoppingCriteriaSub` is a noop here, because `__call__` doesn't ever return `True`. But even if it did that would not be enough for the result we'd probably want here, see the discussion here https://github.com/huggingface/transformers/issues/22340. For context trlX has `stop_sequences` argument which suits this purpose, but it works in post-generation manner